api/v1alpha1/validation/securitypolicy_validate.go

```diff
@@ -53,13 +58,28 @@ func validateSecurityPolicySpec(spec *egv1a1.SecurityPolicySpec) error {
 		return utilerrors.NewAggregate(errs)
 	}
 
-	if err := ValidateJWTProvider(spec.JWT.Providers); err != nil {
+	if err := ValidateAPIKeyAuth(spec.APIKeyAuth); err != nil {
 		errs = append(errs, err)
 	}
-
 	return utilerrors.NewAggregate(errs)
 }
 
+func ValidateAPIKeyAuth(p *egv1a1.APIKeyAuth) error {
+	if p == nil {
+		return nil
+	}
+
+	for _, keySource := range p.ExtractFrom {
```

Comment:
this is a good case where this can be moved to CEL( no action needed here,  just wanted to highlight that)

Reply:
Ah thanks for clarifying